mac/mcompile/keymap.h

```diff
@@ -0,0 +1,549 @@
+#pragma once
+#ifndef KEYMAP_H
+#define KEYMAP_H
+
+#include <iostream>
+#include <Carbon/Carbon.h>
+#include <string>
+#include <vector>
+
+#include "../../common/include/km_u16.h"
+
+enum ShiftState {
+  Base         = 0,                   // 0
+  Shft         = 1,                   // 1
+  Ctrl         = 2,                   // 2
+  ShftCtrl     = Shft | Ctrl,         // 3
+  Menu         = 4,                   // 4 -- NOT USED
+  ShftMenu     = Shft | Menu,         // 5 -- NOT USED
+  MenuCtrl     = Menu | Ctrl,         // 6
+  ShftMenuCtrl = Shft | Menu | Ctrl,  // 7
+  Xxxx         = 8,                   // 8
+  ShftXxxx     = Shft | Xxxx,         // 9
+};
+
+#define VK_SPACE    0x20
```

Comment:
These are another candidate for cleanup in various source files (should be a separate PR on its own)